### ISSUE TITLE
Issue with exifs on samsung devices

### ISSUE THREAD

**Issue by iovorobiev** [jdamcd/android-crop#117]
There are some issues with crop on samsung devices when doing non-square crop. It caused by rotating matrix by -exif_rotate value and than adding exif. Suppose I want to crop picture to get 16:9 rectangle and then set it to ImageView by using standard setImageUri() method.
![cropping](https://cloud.githubusercontent.com/assets/1309921/7632842/25fbd8f6-fa83-11e4-8641-2efe8fde710d.png)

then I getting this

![cropped](https://cloud.githubusercontent.com/assets/1309921/7632878/61b269c8-fa83-11e4-8c82-1afc68d3d8dc.png)

If I will use a library which reads exif and applies rotating then I getting this

![with_glide_2](https://cloud.githubusercontent.com/assets/1309921/7632962/c7691fbe-fa83-11e4-9746-3a323f1f2ea4.png)

So, the main issue, is when we apply exif to cropped picture, we rotate the whole bitmap and change it's dimensions from 16:9 to 9:16. Also, if we not apply exif parameters when setting picture to image view the picture will be rotated by minus exif rotation angle and stretched to given dimensions. I've fixed this this issue in mine pull request https://github.com/jdamcd/android-crop/pull/116. If you have another solution, please fix it.

**Comment by jdamcd** [comment]
Are you using the same `Uri` for input & output?

I'm keen to fix this, but have trouble reproducing the issue. Are you using a specific camera app? Or maybe you could provide an image with problem EXIF data.

**Comment by sridharananthakrishnan** [comment]
I working on camera app ... And I'm using same uri for storing images.
..but what u mean by exifs ... You asked about android interface .??.

**Comment by iovorobiev** [comment]
@jdamcd yes, I am using the same uri for input and output. I use a standard camera app on Samsung Galaxy Note 2. This issue reproduces only on samsung devices.

**Comment by kungfucop** [comment]
we also see the issue on samsung device, is that possible to make a fast fix for that? Thanks!!

**Comment by dkozel** [comment]
I can reliably reproduce this on a Samsung Galaxy S4 with Android 4.4.2.

**Comment by TheWhiteLlama** [comment]
Today I found out that it's possibly related not only to exif itself but also the Uri handling. When opening an image via the "last" section of my system gallery app I could reproduce the bug with the wrong rotation, since the Uri was shortened by the gallery app somehow. After resolving the shortened Uri to the full Uri of the file it worked correctly again. I fixed this in my forked repo and will probably make a pull request during the next couple of days

**Comment by thameemulansari** [comment]
@TheWhiteLlama , Do you have an fix for the issues mentioned through out this thread ? Please provide your branch and code changes . will be helpful.
